src/components/cross-chain.js

```diff
@@ -131,21 +165,69 @@ class CrossChainRecoveryForm extends Component {
       defaultPath: '~/' + fileName
     };
 
-    // Retrieve the desired file path and file name
     const filePath = dialog.showSaveDialog(dialogParams);
     if (!filePath) {
       // TODO: The user exited the file creation process. What do we do?
       return;
     }
 
     try {
-      fs.writeFileSync(filePath, JSON.stringify(fileData, null, 4), 'utf8');
+      fs.writeFileSync(filePath, JSON.stringify(transaction, null, 4), 'utf8');
+    } catch (err) {
+      console.log('error saving', err);
+      this.setState({ error: 'There was a problem saving your recovery file. Please try again.' });
+    }
+  }
+
+  saveMultipleTransactions = async () => {
+    const transactions = this.state.recoveryTxs;
+
+    const zip = new jszip();
+
+    _.forEach(transactions, (transaction, index) => {
+      let fileName;
+
+      if (this.state.signed) {
+        fileName = `${transaction.sourceCoin}r-${this.state.txids[index].slice(0, 6)}-${moment().format('YYYYMMDD')}.signed.json`;
```

Comment:
We might want to extract fileName calculation to its own method - it's in two places, so if we need to update the file format later on, don't want to update it in two places